### PR TITLE
Feat: By default githubrelease and gittag combined with semver shouldn't return original version

### LINKED ISSUE #803: Semantic Version filter drop "v" prefix
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When parsing version using semantic version, updatecli drop the v prefix such as 

```
######################
# BUMP VENOM VERSION #
######################


SOURCES
=======

latestVersion
-------------
Searching for version matching pattern "*"
✔ Github Release version "1.0.1" found matching pattern "*"
```

### Expected Behavior

It should return 

```
######################
# BUMP VENOM VERSION #
######################


SOURCES
=======

latestVersion
-------------
Searching for version matching pattern "*"
✔ Github Release version "v1.0.1" found matching pattern "*"

```

### Steps To Reproduce

`updatecli diff --confeig updatecli/updatecli.d/venom.yaml `

### Environment

```markdown
- OS: linux
- updatecli: v0.29.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by olblak**
Fix #803 

As Herve suggested, I propose a new flag "originalversion" set to false by default. such as

```
sources:
  latestVersion:
    name: Get latest Updatecli release
    kind: githubrelease
    spec:
      owner: updatecli
      repository: updatecli
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "GITHUB_ACTOR" }}'
      keeporiginalversion: true
      versionfilter:
        kind: semver
```

```

SOURCES
=======

latestVersion
-------------
WARNING: Deprecated behavior detected
|++++
| If the Github Release you are interacting with, doesn't prepend the retrieved version with the character "v" such as "v1.0.0",
| then feel free to add the parameter 'originalVersion: true' to hide this message such as in the following example.
| Otherwise please read this message carefully.
| 
| ----
| sources:
|  		example:
|    		name: Get latest Updatecli release
|    		kind: githubrelease
|    		spec:
|      			owner: updatecli
|      			repository: updatecli
|      			token: '{{ requiredEnv "GITHUB_TOKEN" }}'
|      			username: '{{ requiredEnv "GITHUB_ACTOR" }}'
|      			originalVersion: true
|      			versionfilter:
|        			kind: semver
| ----
| 
| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
| Instead, it should be the role of a "Transformer" rule to modify it.
|
| The Github Release version retrieve using the semantic version rule will stop dropping the "v" if one is specified in the GitHub release tag such as v1.0.0.
| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
| 
| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "originalVersion" to keep the current deprecated behavior until all manifests are updated.
| 
| Please adapt your manifest to this new behavior and then set originalVersion to true to validate that you are manifest is compatible with the new behavior.
| More information on https://github.com/updatecli/updatecli/issues/803
|++++


Searching for version matching pattern "*"
✔ Github Release version "1.0.1" found matching pattern "*"

...
```

This warning will only be displayed for "gittag" and "githubreleas" as those two resources are the only one being affected by this change. the "dockerimage" and "gitea/*" haven't be released yet.

## Test

To test this pull request, you can run the following commands:

```shell
go build -o bin/updatecli
./bin/updatecli diff --config e2e/updatecli.d/success.d/gittag.yaml
./bin/updatecli diff --config e2e/updatecli.d/success.d/githubrelease.yaml

./bin/updatecli diff --config e2e/updatecli.d/warning.d/gittag.yaml
./bin/updatecli diff --config e2e/updatecli.d/warning.d/githubrelease.yaml
go test
```

### CODE REVIEW

File: e2e/updatecli.d/success.d/githubrelease.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sources:
+  latestVersion:
+    name: Get latest Venom release
+    kind: githubrelease
+    spec:
+      owner: updatecli
+      repository: updatecli
+      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
+      originalVersion: true
```
Comment (lemeurherve) [inline e2e/updatecli.d/success.d/githubrelease.yaml:10]: Shouldn't this field name lowercase only?

Also, WDYT of `keeporiginalversion`?

File: pkg/plugins/resources/githubrelease/main.go [hunk 2]
```diff
@@ -50,13 +54,18 @@ func New(spec interface{}) (*GitHubRelease, error) {
 		return &GitHubRelease{}, err
 	}
 
+	if !newSpec.OriginalVersion {
+		logrus.Warningf("%s\n\n", DeprecatedSemverVersionVersion)
```
Comment (lemeurherve) [inline pkg/plugins/resources/githubrelease/main.go:58]:
```suggestion
		logrus.Warningf("%s\n\n", DeprecatedSemverVersionMessage)
```
(same elsewhere)

File: pkg/plugins/resources/gittag/deprecation.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package gittag
+
+var (
+	DeprecatedSemverVersionVersion string = `Deprecated behavior detected
+|++++
+| If the GitTag resource you are interacting with, doesn't prepend the retrieved version with the character "v" such as "v1.0.0",
```
Comment (lemeurherve) [inline pkg/plugins/resources/gittag/deprecation.go:6]:
```suggestion
| If the GitTag resource you are interacting with doesn't prepend the retrieved version with the character "v" such as "v1.0.0",
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+|    		spec:
+|      			originalVersion: true
+|      			versionfilter:
+|        			kind: semver
+| ----
+| 
+| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
+| Instead, it should be the role of a "Transformer" rule to modify it.
+|
+| The Git Tag version retrieve using the semantic version rule will stop dropping the "v" if one is specified in the Git tag such as v1.0.0.
```
Comment (lemeurherve) [inline pkg/plugins/resources/gittag/deprecation.go:24]:
```suggestion
| The Git Tag version retrieval using the semantic version rule will stop dropping the "v" if one is specified in the Git tag such as v1.0.0.
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
+| 
+| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "originalVersion" to keep the current deprecated behavior until all manifests are updated.
+| 
+| Please adapt your manifest to this new behavior and then set originalVersion to true to validate that you are manifest is compatible with the new behavior.
```
Comment (lemeurherve) [inline pkg/plugins/resources/gittag/deprecation.go:29]:
```suggestion
| Please adapt your manifest to this new behavior by adding a transformer and then set originalVersion to true to validate that you are manifest is compatible with the new behavior.
```

A complete example of an adapted manifest here would be nice. (maybe even add a before/after?)

File: e2e/updatecli.d/success.d/githubrelease.yaml (hunk 1 above)
Comment (olblak) [reply e2e/updatecli.d/success.d/githubrelease.yaml:10]: I prefer `keeporiginalversion` which I think is more explicit of what it does

File: pkg/plugins/resources/githubrelease/main.go (hunk 2 above)
Comment (olblak) [reply pkg/plugins/resources/githubrelease/main.go]: This one has been solve another commit. thank you for catching this

File: e2e/updatecli.d/success.d/githubrelease.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+sources:
+  latestVersion:
+    name: Get latest Venom release
```
Comment (olblak) [inline e2e/updatecli.d/success.d/githubrelease.yaml:3]:
```suggestion
    name: Get latest Updatecli release
```

File: e2e/updatecli.d/success.d/githubrelease.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    spec:
+      owner: updatecli
+      repository: updatecli
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GITHUB_ACTOR" }}'
+      keeporiginalversion: true
+      versionfilter:
+        kind: semver
+  regex:
+    name: Get latest Venom release
```
Comment (olblak) [inline e2e/updatecli.d/success.d/githubrelease.yaml:14]:
```suggestion
    name: Get latest Updatecli release
```

File: e2e/updatecli.d/warning.d/githubrelease.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+sources:
+  latestVersion:
+    name: Get latest Venom release
```
Comment (olblak) [inline e2e/updatecli.d/warning.d/githubrelease.yaml:3]:
```suggestion
    name: Get latest Updatecli release
```

File: pkg/plugins/resources/githubrelease/deprecation.go [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+| >                 - trimprefix: v
+| ----
+| 
+| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
+| Instead, it should be the role of a "Transformer" rule to modify it.
+|
+| The Github Release version retrieve using the semantic version rule will stop dropping the "v" if one is specified in the GitHub release tag such as v1.0.0.
+| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
+| 
+| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "originalVersion" to keep the current deprecated behavior until all manifests are updated.
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/deprecation.go:49]:
```suggestion
| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "keeporiginalversion" to keep the current deprecated behavior until all manifests are updated.
```

File: pkg/plugins/resources/githubrelease/deprecation.go [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+| 
+| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
+| Instead, it should be the role of a "Transformer" rule to modify it.
+|
+| The Github Release version retrieve using the semantic version rule will stop dropping the "v" if one is specified in the GitHub release tag such as v1.0.0.
+| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
+| 
+| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "keeporiginalversion" to keep the current deprecated behavior until all manifests are updated.
+| 
+| Please adapt your manifest to this new behavior and then set originalVersion to true to validate that you are manifest is compatible with the new behavior.
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/deprecation.go:51]:
```suggestion
| Please adapt your manifest to this new behavior and then set "keeporiginalversion" to true to validate that you are manifest is compatible with the new behavior.
```

File: pkg/plugins/resources/githubrelease/main.go [hunk 11]
```diff
@@ -21,13 +22,16 @@ type Spec struct {
 	Username string `yaml:",omitempty" jsonschema:"required"`
 	// VersionFilter provides parameters to specify version pattern and its type like regex, semver, or just latest.
 	VersionFilter version.Filter `yaml:",omitempty"`
+	// KeepOriginalVersion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
+	KeepOriginalVersion bool
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/main.go:26]:
```suggestion
	KeepOriginalVersion bool `yaml:",omitempty"`
```

File: pkg/plugins/resources/gittag/main.go [hunk 12]
```diff
@@ -18,6 +19,8 @@ type Spec struct {
 	VersionFilter version.Filter `yaml:",omitempty"`
 	// Message associated to the git tag
 	Message string `yaml:",omitempty"`
+	// Keeporiginalversion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
+	KeepOriginalVersion bool
```
Comment (olblak) [inline pkg/plugins/resources/gittag/main.go:23]:
```suggestion
	KeepOriginalVersion bool  `yaml:",omitempty"`
```

File: updatecli/updatecli.d/venom.yaml [hunk 13]
```diff
@@ -21,6 +21,7 @@ sources:
       repository: venom
       token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
       username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
+      originalVersion: true
```
Comment (olblak) [inline updatecli/updatecli.d/venom.yaml:24]:
```suggestion
      keeporiginalversion: true
```

File: pkg/plugins/resources/githubrelease/deprecation.go [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+| > sources:
+| >     example:
+| >         name: Get latest Updatecli release
+| >         kind: githubrelease
+| >         spec:
+| >             owner: updatecli
+| >             repository: updatecli
+| >             token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+| >             username: '{{ requiredEnv "GITHUB_ACTOR" }}'
+| >             originalVersion: true
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/deprecation.go:36]:
```suggestion
| >             keeporiginalversion: true
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+| ----
+|
+| After:
+| ----
+| > sources:
+| >     example:
+| >         name: Get latest Updatecli release
+| >         kind: gittag
+| >    	    spec:
+| >             originalVersion: true
```
Comment (olblak) [inline pkg/plugins/resources/gittag/deprecation.go:28]:
```suggestion
| >             keeporiginalversion: true
```

File: pkg/plugins/resources/githubrelease/deprecation.go [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+| >                 kind: semver
+| >             transformers:
+| >                 - trimprefix: v
+| ----
+| 
+| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
+| Instead, it should be the role of a "Transformer" rule to modify it.
+|
+| The Github Release version retrieve using the semantic version rule will stop dropping the "v" if one is specified in the GitHub release tag such as v1.0.0.
+| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/deprecation.go:47]:
```suggestion
| This bug has been around for so long that we saw a significant amount of Updatecli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+| >                 - trimprefix: v
+| ----
+| 
+| In Updatecli, we consider that information retrieved from a "Source" shouldn't be altered.
+| Instead, it should be the role of a "Transformer" rule to modify it.
+|
+| The Git Tag version retrieval using the semantic version rule will stop dropping the "v" if one is specified in the Git tag such as v1.0.0.
+| This bug has been around for so long that we saw a significant amount of Updateclli manifest written in a way that we can't automatically fix by running "updatecli manifest upgrade"
+| 
+| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "originalVersion" to keep the current deprecated behavior until all manifests are updated.
```
Comment (olblak) [inline pkg/plugins/resources/gittag/deprecation.go:41]:
```suggestion
| As we take seriously backward compatibility for Updatecli manifest, we introduce the parameter "keeporiginalversion" to keep the current deprecated behavior until all manifests are updated.
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+| 
+| Please adapt your manifest to this new behavior by adding a transformer and then set originalVersion to true to validate that you are manifest is compatible with the new behavior.
```
Comment (olblak) [inline pkg/plugins/resources/gittag/deprecation.go:43]:
```suggestion
| Please adapt your manifest to this new behavior by adding a transformer and then set keeporiginalVersion to true to validate that you are manifest is compatible with the new behavior.
```

File: pkg/plugins/resources/githubrelease/deprecation.go [hunk 19]
```diff
@@ -0,0 +1,55 @@
+package githubrelease
+
+var (
+	DeprecatedSemverVersionMessage string = `Deprecated behavior detected
+|++++
+| If the Github Release you are interacting with, doesn't prepend the retrieved version with the character "v" such as "v1.0.0",
+| then feel free to add the parameter 'originalVersion: true' to hide this message such as in the following example.
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/deprecation.go:7]:
```suggestion
| then feel free to add the parameter 'keeporiginalVersion: true' to hide this message such as in the following example.
```

File: pkg/plugins/resources/gittag/deprecation.go [hunk 20]
```diff
@@ -0,0 +1,47 @@
+package gittag
+
+var (
+	DeprecatedSemverVersionMessage string = `Deprecated behavior detected
+|++++
+| If the GitTag resource you are interacting with doesn't prepend the retrieved version with the character "v" such as "v1.0.0",
+| then feel free to add the parameter 'originalVersion: true' to hide this message such as in the following example.
```
Comment (olblak) [inline pkg/plugins/resources/gittag/deprecation.go:7]:
```suggestion
| then feel free to add the parameter 'keeporiginalVersion: true' to hide this message such as in the following example.
```

File: pkg/plugins/resources/githubrelease/main.go [hunk 21]
```diff
@@ -21,13 +22,16 @@ type Spec struct {
 	Username string `yaml:",omitempty" jsonschema:"required"`
 	// VersionFilter provides parameters to specify version pattern and its type like regex, semver, or just latest.
 	VersionFilter version.Filter `yaml:",omitempty"`
+	// KeepOriginalVersion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
```
Comment (olblak) [inline pkg/plugins/resources/githubrelease/main.go:25]:
```suggestion
	// [S] KeepOriginalVersion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
```

File: pkg/plugins/resources/gittag/main.go [hunk 22]
```diff
@@ -18,6 +19,8 @@ type Spec struct {
 	VersionFilter version.Filter `yaml:",omitempty"`
 	// Message associated to the git tag
 	Message string `yaml:",omitempty"`
+	// Keeporiginalversion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
```
Comment (olblak) [inline pkg/plugins/resources/gittag/main.go:22]:
```suggestion
	// [S] Keeporiginalversion is an ephemeral parameters. cfr https://github.com/updatecli/updatecli/issues/803
```

**Comment by olblak** [comment]
Closing in favor of #816

**Pull request closed without merging**